client/lib/products-list/index.js

```diff
@@ -99,14 +98,19 @@ ProductsList.prototype.fetch = function() {
 			this.emit( 'change' );
 
 			if ( typeof localStorage !== 'undefined' ) {
-				localStorage.setItem( 'ProductsList', JSON.stringify( productsList ) );
+				try {
+					localStorage.setItem( 'ProductsList', JSON.stringify( productsList ) );
```

Comment:
fixes an issue i noticed while testing how nock works. If localstorage is available, but broken, we should catch the error and keep going.